aws-py-apigatewayv2-eventbridge/__main__.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2016-2021, Pulumi Corporation.  All rights reserved.
+
+import json
+import pulumi
+import pulumi_aws as aws
+
+# Create an HTTP API.
+api = aws.apigatewayv2.Api("example",
+    protocol_type="HTTP"
+)
+
+# Create a stage and set it to deploy automatically.
+stage = aws.apigatewayv2.Stage("stage",
+    api_id=api.id,
+    name=pulumi.get_stack(),
+    auto_deploy=True
+);
+
+# Create an event bus.
+bus = aws.cloudwatch.EventBus("bus")
+
+# Create an event rule to watch for events.
+rule = aws.cloudwatch.EventRule("rule",
+    event_bus_name=bus.name,
+    event_pattern=json.dumps({"source": ["my-event-source"]})
+)
+
+# Define a policy granting API Gateway permission to publish to EventBridge.
+api_gateway_role = aws.iam.Role("api-gateway-role",
+    assume_role_policy=json.dumps({
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Action": "sts:AssumeRole",
+                "Effect": "Allow",
+                "Principal": {
+                    "Service": "apigateway.amazonaws.com",
+                },
+            },
+        ],
+    }),
+    managed_policy_arns=[
+        "arn:aws:iam::aws:policy/AmazonEventBridgeFullAccess",
+    ],
+);
+
+# Create an API Gateway integration to forward requests to EventBridge.
+integration = aws.apigatewayv2.Integration("integration",
+    api_id=api.id,
+
+    # The integration type and subtype.
+    integration_type="AWS_PROXY",
+    integration_subtype="EventBridge-PutEvents",
+    credentials_arn=api_gateway_role.arn,
+
+    # The body of the request to be sent to EventBridge. Note the
+    # event source matches pattern defined on the EventRule, and the
+    # Detail expression, which just forwards the body of the original
+    # API Gateway request (i.e., the uploaded document).
+    request_parameters={
+        "EventBusName": bus.name.apply(lambda name: name),
+        "Source": "my-event-source",
+        "DetailType": "my-detail-type",
+        "Detail": "$request.body",
```

Comment:
Should this be `$request.body.name` based on https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-parameter-mapping.html#http-api-mapping-request-parameters ?

(Also, I _am_ reading that right that you don't need curly braces since it's only one variable, right? It's kinda itchy in Python 😆  )

Reply:
> Should this be $request.body.name

I'm passing the whole request body through in this case, so I think this is okay.

 > you don't need curly braces

I don't think I follow. Which braces do you mean? The ones around these four properties? If I take those out, I get this:

![image](https://user-images.githubusercontent.com/274700/149210286-663f4f93-6847-4f3c-adb3-c501290a714b.png)

... but if there's something weird-looking or otherwise non-idiomatic here, definitely let me know and I'm happy to change.

Reply:
Ah, I see. Yes:

> I think, if I'm reading everything right, that string gets passed as is to AWS, which then does substitution, rather than Python doing the substitution.

That's correct. This is just a raw string that's passed to AWS, and in this case, it just means "the full request body". You can see it referenced this way under Mapping Request Parameters:

https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-aws-services.html

![image](https://user-images.githubusercontent.com/274700/149216190-6a9f707c-6ae4-4070-b7a9-306cef99aa2a.png)